GVFS/GVFS.Virtualization/FileSystemCallbacks.cs

```diff
@@ -763,6 +785,26 @@ private FileSystemTaskResult ExecuteBackgroundOperation(FileSystemTask gitUpdate
             return result;
         }
 
+        private FileSystemTaskResult TryRemoveModifiedPath(string virtualPath, bool isFolder)

```

Comment:
Should this method also be called for renames (with the old path when the old path is in `newlyCreatedFileAndFolderPaths`)?

If so, this might be a reason to plumb that through the Mac callbacks as well (as a separate follow up issue, CC: @sanoursa for his thoughts).

Reply:
If we do this then shouldn't the new path be added to the `newlyCreatedFileAndFolderPaths` so that if a file is created, renamed and then deleted nothing will be left in the modified paths?

Reply:
That's sounds like a good idea as well.  When I left my original comment I was thinking about the common tmp+rename pattern:

1. Create tmp file
2. Write to tmp file
3. Rename+overwrite to final path

But you're right that after step #3 that file could be deleted (and if it were new to start with it's something else we could remove from modified paths).